api/algorithm_site_instance.py

```diff
@@ -143,6 +145,9 @@ def __init__(
         self.ids = recrudescence_utils.get_sample_ids(genotypedata_RR, 'Day 0')
         self.locinames = pd.unique(genotypedata_RR.columns[1:].str.split("_").str[0])
 
+        if len(self.locinames) > len(locirepeats):
+            raise ValueError("Loci repeats has an insufficient number of entries")
```

Comment:
Could you make a custom exception for improper loci repeats in this file, and then use that? I know that's annoying, but ValueErrors can be thrown by other things that might lead to confusing error messages about loci repeats that have nothing to do with the loci

Reply:
Change made, and no problem! Completely see the point about clarity, and the effort to implement was minimal. With this change, I think it should be ready for re-review